channels_redis/core.py

```diff
@@ -207,15 +220,23 @@ async def send(self, channel, message):
 
         # Check the length of the list before send
         # This can allow the list to leak slightly over capacity, but that's fine.
-        if await connection.zcount(channel_key, "-inf", "+inf") >= self.get_capacity(
-            channel
-        ):
+        current_length = await connection.zcount(channel_key, "-inf", "+inf")
+
+        if current_length + len(messages) > self.get_capacity(channel):
             raise ChannelFull()
 
         # Push onto the list then set it to expire in case it's not consumed
-        await connection.zadd(channel_key, {self.serialize(message): time.time()})
+        await connection.zadd(channel_key, mapping)
         await connection.expire(channel_key, int(self.expiry))
 
+    def _parse_messages(self, message):
+        """
+        Convert a passed message arg to a tuple of messages.
+        """
+        if not isinstance(message, dict) and hasattr(message, "__iter__"):
+            return tuple(message)
+        return (message,)
```

Comment:
Seems a shame to consume the iterator only to loop over it again in the `for message in messages` loop. 🤔

Reply:
Good point, will rewrite that